### PR TITLE
[AGENT-4613] Add Airflow DAG predictions examples with Scoring Code

### PR DESCRIPTION
**Pull request by oleksandr-saienko**
# This repository is public. Do not put here any private DataRobot or customer's data: code, datasets, model artifacts, .etc.

## Summary
Added Airflow DAG predictions examples with Scoring Code
Added Operators for scoring code downloading from deployment and from model

## Rationale
Users should be able to use Airflow with scoring code to run predictions

### CODE REVIEW

**Comment by oleksandr-saienko** [comment]
Jenkins run

**Comment by oleksandr-saienko** [comment]
Jenkins run

File: datarobot_provider/example_dags/download_scoring_code_from_deployment_dag.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        # The next scoring code example is only for demo and testing proposes,
+        # should be used only with a very small datasets (not higher than a few megabytes).
+        # To run this example, DataRobot Prediction Library (https://datarobot.github.io/datarobot-predict)
+        # should be installed first.
+
+        input_file = "/.../input_file.csv"
+        output_file = "/.../output_predictions.csv"
+
+        import pandas as pd
+        from datarobot_predict.scoring_code import ScoringCodeModel
```
Comment (andrius-senulis) [inline datarobot_provider/example_dags/download_scoring_code_from_deployment_dag.py:57]: why do you need to have imports in the function rather than top of module?

File: tests/unit/operators/test_scoring_code.py [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+    mocker.patch.object(dr.Deployment, "download_scoring_code")
+
+    operator = DownloadDeploymentScoringCodeOperator(
+        task_id="download_scoring_code_from_deployment",
+        deployment_id="deployment-id",
+        base_path=scoring_code_filepath,
+    )
+
+    with pytest.raises(ValueError):
```
Comment (andrius-senulis) [inline tests/unit/operators/test_scoring_code.py:78]: you may also check that the right error is raised with:
```
with pytest.raises(ValueError, match="Invalid or missing base path value"):
```

File: tests/unit/operators/test_scoring_code.py
Comment (andrius-senulis) [inline tests/unit/operators/test_scoring_code.py:1]: I would also test that it works as expected when `DownloadDeploymentScoringCodeOperator`  is called with `base_path=None` but `context["params"].get("scoring_code_filepath")` is set. Also for model operator.

**Review by andrius-senulis — APPROVED** [review]
A few minor comments.

File: datarobot_provider/example_dags/download_scoring_code_from_deployment_dag.py (hunk 1 above)
Comment (oleksandr-saienko) [reply datarobot_provider/example_dags/download_scoring_code_from_deployment_dag.py:57]: its just a sample DAG, and datarobot_predict currently is not part of airflow provider so I decided to put all required stuff related to scoring inside Python operator
